File: src/main/java/net/igneo/icv/sound/tickable/FollowingSound.java
package net.igneo.icv.sound.tickable;

import net.minecraft.client.resources.sounds.AbstractTickableSoundInstance;
import net.minecraft.sounds.SoundEvent;
import net.minecraft.sounds.SoundSource;
import net.minecraft.util.RandomSource;
import net.minecraft.world.entity.Entity;

public class FollowingSound extends AbstractTickableSoundInstance {
    Entity entity;
    public FollowingSound(SoundEvent sound, Entity entity) {
        super(sound, SoundSource.PLAYERS, RandomSource.create()); // Or AMBIENT if you prefer
        this.entity = entity;
        this.delay = 0;
        this.x = entity.getX();
        this.y = entity.getY();
        this.z = entity.getZ();
    }

    @Override
    public void tick() {
        this.x = entity.getX();
        this.y = entity.getY();
        this.z = entity.getZ();
    }
}

File: src/main/java/net/igneo/icv/sound/tickable/FollowingSound.kt
package net.igneo.icv.sound.tickable

import net.minecraft.client.resources.sounds.AbstractTickableSoundInstance
import net.minecraft.sounds.SoundEvent
import net.minecraft.sounds.SoundSource
import net.minecraft.util.RandomSource
import net.minecraft.world.entity.Entity

class FollowingSound(sound: SoundEvent, var entity: Entity) :
    AbstractTickableSoundInstance(sound, SoundSource.PLAYERS, RandomSource.create()) {
    init {
        this.delay = 0
        this.x = entity.x
        this.y = entity.y
        this.z = entity.z
    }

    override fun tick() {
        this.x = entity.x
        this.y = entity.y
        this.z = entity.z
    }
}
